### PR TITLE
4.08 WIP

### PR DESCRIPTION
**Pull request by hhugo**
Initial work to support OCaml 4.08.

Should fix https://github.com/janestreet/base/issues/66

### CODE REVIEW

**Comment by hhugo** [comment]
cc @trefis

File: src/import0.ml [hunk 1]
```diff
@@ -9,124 +9,65 @@ include
...
-  (** @canonical Caml.Arg *)
-  module Arg       = Caml.Arg
+  include (Caml : (module type of struct include Caml end
+                   with module Array := Caml.Array
+                    and module Bytes := Caml.Bytes
+                    and module Hashtbl := Caml.Hashtbl
+                    and module List := Caml.List
+                    and module Map := Caml.Map
+                    and module Set := Caml.Set
+                    and module String := Caml.String))
```
Comment (hhugo) [inline src/import0.ml]: This might generate bad documentation.

**Comment by hhugo** [comment]
This is being reviewed internally at janestreet.

**Comment by XVilka** [comment]
Will it be the part of new release? Seems 4.08 is almost ready.

**Comment by hhugo** [comment]
Base v0.12.2 has been released already.
https://github.com/ocaml/opam-repository/pull/14177

**Pull request closed without merging**